config/routes.rb

```diff
@@ -7,6 +7,12 @@
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
   namespace :api do
     namespace :v1, defaults: { format: :json } do # TODO: implement new routes later
+      resource :carts, only: [:show] do
+        collection do
+          post :add_product
+          post :remove_product
```

Comment:
The `remove_product` endpoint updates the cart, right? Can we use PUT instead of POST?

Reply:
It has been decided to use the `Post` method for these endpoints because the business logic involves not only modifying the `Cart` model but its intermediate table with the `Product` model (`CartProduct`) to create, update or delete. Therefore, for managing resources, it is more appropriate to use the `Post` verb.

Reply:
Updated to `PUT`